calendar/experiments/calendar/parent/ext-calendar-provider.js

```diff
@@ -258,6 +258,34 @@ class ExtCalendarProvider extends cal.provider.BaseClass {
   }
 }
 
+class ExtFreeBusyProvider {
+  QueryInterface = ChromeUtils.generateQI(["calIFreeBusyProvider"]);
+
+  constructor(fire) {
+    this.fire = fire;
+  }
+
+  async getFreeBusyIntervals(aCalId, aRangeStart, aRangeEnd, aBusyTypes, aListener) {
+    try {
+      const TYPE_MAP = {
+        free: Ci.calIFreeBusyInterval.FREE,
+        busy: Ci.calIFreeBusyInterval.BUSY,
+        unavailable: Ci.calIFreeBusyInterval.BUSY_UNAVAILABLE,
+        tentative: Ci.calIFreeBusyInterval.BUSY_TENTATIVE,
+      };
+      let attendee = aCalId.replace(/^mailto:/, "");
+      let start = aRangeStart.icalString;
+      let end = aRangeEnd.icalString;
+      let types = ["free", "busy", "unavailable", "tentative"].filter((type, index) => aBusyTypes & 1 << index);
+      let results = await this.fire.async({ attendee, start, end, types });
+      aListener.onResult({ status: Cr.NS_OK }, results.map(interval => new cal.provider.FreeBusyInterval(aCalId, TYPE_MAP[interval.type], cal.createDateTime(interval.start), cal.createDateTime(interval.end))));
```

Comment:
If you are making changes anyway I'd appreciate if you could wrap the line a bit here. We should probably add max length to eslint.

Reply:
Done.